src/parsers/texttracks/webvtt/html/create_style_attribute.ts

```diff
@@ -0,0 +1,219 @@
+  const hasSettings = settings !== undefined && objectValues(settings).length !== 0;
+  if (!hasSettings) {
+    return "text-align:center";
+  }
+
+  return (
+    "position: absolute;" +
+    "margin: 0;" +
+    `${getTransformStyle(settings)}` +
+    `${getSizeStyle(settings.size)}` +
+    `${getPositionStyle(settings)}` +
+    `${getLineStyle(settings.line)}` +
+    `${getAlignStyle(settings.align)}`
+  );
+};
+
+enum PositionAlignment {
+  LINE_LEFT = "line-left",
+  CENTER = "center",
+  LINE_RIGHT = "line-right",
+}
+
+enum Align {
+  LEFT = "left",
+  CENTER = "center",
+  RIGHT = "right",
+}
+
+enum LineAlignment {
+  START = "start",
+  CENTER = "center",
+  END = "end",
+}
+
+/**
+ * Used to shift cue box origin in order to define
+ * what the position and line values are relating to.
+ */
+const getTransformStyle = (settings: Partial<Record<string, string>>) => {
+  return `transform: translate(${getTransformX(settings)}%,${getTransformY(settings.line)}%);`;
+};
+
+const getTransformX = (settings: Partial<Record<string, string>>) => {
+  const positionAlignmentMap = {
+    [PositionAlignment.LINE_LEFT]: 0,
+    [PositionAlignment.CENTER]: -50,
+    [PositionAlignment.LINE_RIGHT]: -100,
+  };
+
+  const positionAlignment = settings.position !== undefined ?
+    getPositionAlignment(settings.position) :
+    null;
+
+  if (positionAlignment !== null) {
+    return positionAlignmentMap[positionAlignment];
+  }
+
+  const alignMap = {
+    [Align.LEFT]: 0,
+    [Align.CENTER]: -50,
+    [Align.RIGHT]: -100,
+  };
+
+  const align = settings.align !== undefined ?
+    getAlign(settings.align)
+    : Align.CENTER;
+
+  return alignMap[align];
+};
+
+const getTransformY = (lineSetting: string | undefined) => {
+  const lineAlignmentMap = {
+    [LineAlignment.START]: 0,
+    [LineAlignment.CENTER]: -50,
+    [LineAlignment.END]: -100,
+  };
+
+  if (lineSetting === undefined) {
+    return lineAlignmentMap[LineAlignment.START];
+  }
+
+  const lineAlignment = getLineAlignment(lineSetting);
+
+  return lineAlignment !== null ?
+    lineAlignmentMap[lineAlignment] :
+    lineAlignmentMap[LineAlignment.START];
+};
+
+const getPositionAlignment = (positionSetting: string): PositionAlignment | null => {
+  const positionRegex = /,(line-left|line-right|center)/;
+  const matches = positionRegex.exec(positionSetting);
+
+  if (!Array.isArray(matches) || matches.length < 2) {
+    return null;
+  }
+
+  return matches[1] as PositionAlignment;
+};
+
+const getLineAlignment = (lineSetting: string): LineAlignment | null => {
+  const positionRegex = /,(start|center|end)/;
+  const matches = positionRegex.exec(lineSetting);
+
+  if (!Array.isArray(matches) || matches.length < 2) {
+    return null;
+  }
+
+  return matches[1] as LineAlignment;
+};
+
+const getSizeStyle = (sizeSetting: string | undefined) => {
+  const defaultSize = 100;
+  const size = getPercentageValueOrDefault(sizeSetting, defaultSize);
+  return `width:${size}%;`;
```

Comment:
Hi !
Something confused me in the PR. In order to shift the cues, you use a translation whose parameters are in percentage of the cue box dimensions. However, by default, you set the cue box width to 100% of the parent. It seems that the parent here is the video viewport.

Isn't it a risk that the cue may shift further than expected ? For example, by default, the cue would not be centered on the X axis, but shifting by 50% of the viewport to the left (so being at start of X axis on viewport).

Reply:
According to the [specification](https://www.w3.org/TR/webvtt1/#webvtt-cue-size) the size should default to 100%. Note also that the position defaults to 50%, meaning that the cue box will be placed in the center under the default circumstances. I see what you're meaning though, but I struggle to find a case where this leads to an unexpected placement of the cue. Do you have another example?

Reply:
If i'm not wrong, the default style for the p element becomes :
```
    margin: 0;
    position: absolute;
    transform: translate(-50%, 0);
    width: 100%;
    left: 50%;
```

I made a test on a blank page with the p element and the region parent element, as it is styled in the RxPlayer core, and it seems that the subtitle is located at the bottom left, by default.

Reply:
I've only considered the the video viewport being the cue box here, and as I understand the specification that should be the correct default style in that case. When providing a [textTrackElement](https://github.com/canalplus/rx-player/blob/ffbca0935ffcb2a0c2e9c02aa791d6367f16cc9f/doc/api/loadVideo_options.md#prop-textTrackElement) with the the same size as the video viewport, it should be placed in the center - as far as I can see. 

To consider custom [regions](https://www.w3.org/TR/webvtt1/#regions) I think there's more changes needed.

Reply:
Ok, i understand that the default width may be 100%.
However, it still do not get why we use the translate. The 50% translation is made against the total width of the div, so on the width of the video viewport. So, the text content is stuck to the left, instead of being centered. I'll see what we can do.

Reply:
The translation is needed to correctly handle what the position (the left value styling wise) refers to, which is the [position alignment](https://www.w3.org/TR/webvtt1/#webvtt-cue-position-alignment). Without it, the position would always refer to the left side of the cue box, which is not what the spec says. 

When using the video viewport as the cue box, the translation is done against the total width of the video viewport - which in that case is expected. And again, the position defaults to 50% which places the text in the center in that scenario. 

But okay, iterate on your end and let me know if you want any input!